assert/assertions_test.go

```diff
@@ -2189,3 +2190,22 @@ func Test_validateEqualArgs(t *testing.T) {
 		t.Error("nil functions are equal")
 	}
 }
+
+func Test_truncatingFormat(t *testing.T) {
+
+    original := strings.Repeat("a", bufio.MaxScanTokenSize-102)
+    result := truncatingFormat(original)
+    if fmt.Sprintf("%#v", original) != result {
+        t.Error("string should not be truncated")
+    }
+
+    original = original +"x"
+    result = truncatingFormat(original)
+    if fmt.Sprintf("%#v", original) == result {
+        t.Error("string should have been truncated.")
+    }
+
+    if !strings.HasSuffix(result, "<... truncated>") {
```

Comment:
With the principle of dogfooding, you could use the `Contains()` function but I understand that it doesn't exactly cover the kind of comparison we want. Happy for you to make a call on this